include/triton/Dialect/Triton/IR/TritonOps.td

```diff
@@ -565,7 +565,7 @@ def TT_DotOp : TT_Op<"dot", [Pure,
       TT_FpIntTensor:$a,
       TT_FpIntTensor:$b,
       TT_FpIntTensor:$c,
-      BoolAttr:$allowTF32,
+      StrAttr:$f32Backend,
```

Comment:
nit: making it an enum would be a bit less error-prone. You can use `I32EnumAttr` for that

Reply:
If we use an enum, do we have to list all possible values in TritonOps.td?  That doesn't work well for out-of-tree backends.

(I think it would be good to have the list of valid strings written down somewhere in the nvidia backend, though.)

Reply:
correct the enum has to be in TritonOps.td, it still feels better than doing string checks? I believe we could decouple it using interfaces but that's probably an overkill to do it right now. Even if we have a lit of valid strings, it feels easy to make a typo when checkin the value. 
I'd be fine with the possible constants be declared in a header as constexpr and used instead of literal strings

Reply:
> correct the enum has to be in TritonOps.td, it still feels better than doing string checks?

I just thought we had it as a goal to make it possible for people to write out-of-tree backends?  If you think that's not relevant here, then definitely enums would be better.

> I'd be fine with the possible constants be declared in a header as constexpr and used instead of literal strings

The problem is there's no way to enforce this, and people will use strings anyway.  (And indeed the IR uses a string, so that's the place where we're most likely to have a typo.)  So I think there's an argument for leaning into the strings in the C++ and accepting that we need to write tests.  At least, that is my experience with this sort of thing from XLA, where it's used a lot.

Again I do think it needs to be documented in the backend which strings are acceptable.  Right now I only see it in the frontend.  I also think (orthogonal to all this) that we need a check during lowering that we know how to lower the string we got (i.e. check for invalid strings during lowering -- I don't see this anywhere) in order to catch typos in the IR.

Reply:
I went with the enum implementation, as we already use enums in quite a few places. Going from enums to any other representation in the future should be trivial tho.